pkg/nsx/services/inventory/build.go

```diff
@@ -200,3 +200,58 @@ func (s *InventoryService) BuildNamespace(namespace *corev1.Namespace) (retry bo
 	}
 	return
 }
+
+func (s *InventoryService) BuildService(service *corev1.Service) (retry bool) {
+	log.Info("Building Service", "Service", service.Name, "Namespace", service.Namespace)
+	retry = false
+
+	preContainerApplication := s.ApplicationStore.GetByKey(string(service.UID))
+	if preContainerApplication != nil {
+		preContainerApplication = *preContainerApplication.(*containerinventory.ContainerApplication)
+	}
+
+	namespace := &corev1.Namespace{}
+	err := s.Client.Get(context.TODO(), types.NamespacedName{Name: service.Namespace}, namespace)
+	if err != nil {
+		retry = true
+		log.Error(err, "Failed to get namespace for Service", "Service", service)
+		return
+	}
+
+	netStatus := NetworkStatusHealthy
+	status := InventoryStatusUp
+	switch service.Spec.Type {
+	case corev1.ServiceTypeLoadBalancer:
+		if len(service.Status.LoadBalancer.Ingress) == 0 {
+			netStatus = NetworkStatusUnhealthy
+			status = InventoryStatusDown
+		}
+	case corev1.ServiceTypeClusterIP:
+		if service.Spec.ClusterIP == corev1.ClusterIPNone {
+			netStatus = NetworkStatusUnhealthy
+			status = InventoryStatusDown
+		}
+	}
```

Comment:
Just to confirm if we would like to just check the IPs for service network status. I found antrea will check the endpoints and in our doc we mention we may check error annotation added by ncp. Would we drop these plans or we will add them in the future?